java.lang.UnsupportedOperationException using @RolesAllowed and @SecurityRequirement

While using @RolesAllowed and @SecurityRequirement in a Rest Endpoint:

```
@SecurityRequirement(name = "myrealm")
@RolesAllowed({ SecurityRoles.ADMINISTRATOR, SecurityRoles.APPMANAGER })
public class AppVersionsManagementRest {
```


I get this exception on boot:

```
Caused by: java.lang.UnsupportedOperationException
at java.base/java.util.AbstractList.add(AbstractList.java:153)
at java.base/java.util.AbstractList.add(AbstractList.java:111)
at io.smallrye.openapi.runtime.scanner.processor.JavaSecurityProcessor.addRolesAllowed(JavaSecurityProcessor.java:142)
at io.smallrye.openapi.runtime.scanner.processor.JavaSecurityProcessor.processSecurityRolesForMethodOperation(JavaSecurityProcessor.java:98)
at io.smallrye.openapi.runtime.scanner.processor.JavaSecurityProcessor.processSecurityRoles(JavaSecurityProcessor.java:39)
at io.smallrye.openapi.jaxrs.JaxRsAnnotationScanner.processResourceMethod(JaxRsAnnotationScanner.java:493)
at io.smallrye.openapi.jaxrs.JaxRsAnnotationScanner.lambda$processResourceMethods$0(JaxRsAnnotationScanner.java:289)
```

That is because when the @SecurityRequirement doesn't have any scope, addScheme uses an immutable collection here (L38):
https://github.com/smallrye/smallrye-open-api/blob/23842c0c95a937ba2bc1fc5e7789d636a7e8fbac/core/src/main/java/io/smallrye/openapi/api/models/security/SecurityRequirementImpl.java#L35-L42

But addRolesAllowed tries to add some role to it (L142):

https://github.com/smallrye/smallrye-open-api/blob/23842c0c95a937ba2bc1fc5e7789d636a7e8fbac/core/src/main/java/io/smallrye/openapi/runtime/scanner/processor/JavaSecurityProcessor.java#L135-L145

--- a/core/src/main/java/io/smallrye/openapi/api/models/security/SecurityRequirementImpl.java
+++ b/core/src/main/java/io/smallrye/openapi/api/models/security/SecurityRequirementImpl.java
@@ -1,5 +1,6 @@
 package io.smallrye.openapi.api.models.security;
 
+import java.util.ArrayList;
 import java.util.Collections;
 import java.util.LinkedHashMap;
 import java.util.List;
@@ -22,9 +23,9 @@ public class SecurityRequirementImpl extends LinkedHashMap<String, List<String>>
     @Override
     public SecurityRequirement addScheme(String securitySchemeName, String scope) {
         if (scope == null) {
-            this.put(securitySchemeName, Collections.emptyList());
+            addScheme(securitySchemeName);
         } else {
-            this.put(securitySchemeName, Collections.singletonList(scope));
+            addScheme(securitySchemeName, new ArrayList<>(Collections.singletonList(scope)));
         }
         return this;
     }
@@ -35,7 +36,7 @@ public class SecurityRequirementImpl extends LinkedHashMap<String, List<String>>
     @Override
     public SecurityRequirement addScheme(String securitySchemeName, List<String> scopes) {
         if (scopes == null) {
-            scopes = Collections.emptyList();
+            scopes = new ArrayList<>(0);
         }
         this.put(securitySchemeName, scopes);
         return this;
@@ -46,7 +47,7 @@ public class SecurityRequirementImpl extends LinkedHashMap<String, List<String>>
      */
     @Override
     public SecurityRequirement addScheme(String securitySchemeName) {
-        this.put(securitySchemeName, Collections.emptyList());
+        addScheme(securitySchemeName, (List<String>) null);
         return this;
     }
 
@@ -72,7 +73,7 @@ public class SecurityRequirementImpl extends LinkedHashMap<String, List<String>>
     @Override
     public void setSchemes(Map<String, List<String>> items) {
         this.clear();
-        this.putAll(items);
+        items.forEach(this::addScheme);
     }
 
 }

--- a/core/src/main/java/io/smallrye/openapi/runtime/scanner/processor/JavaSecurityProcessor.java
+++ b/core/src/main/java/io/smallrye/openapi/runtime/scanner/processor/JavaSecurityProcessor.java
@@ -5,6 +5,8 @@ import java.util.Arrays;
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 import org.eclipse.microprofile.openapi.models.OpenAPI;
 import org.eclipse.microprofile.openapi.models.Operation;
@@ -141,12 +143,12 @@ public class JavaSecurityProcessor {
 
             if (requirement.hasScheme(currentSecurityScheme)) {
                 // The name of the declared requirement must match the scheme's name
-                List<String> scopes = requirement.getScheme(currentSecurityScheme);
-                for (String role : roles) {
-                    if (!scopes.contains(role)) {
-                        scopes.add(role);
-                    }
-                }
+                List<String> scopes = Stream.concat(
+                        requirement.getScheme(currentSecurityScheme).stream(),
+                        Arrays.stream(roles))
+                        .distinct()
+                        .collect(Collectors.toList());
+                requirement.addScheme(currentSecurityScheme, scopes);
             }
         }
     }

--- a/extension-jaxrs/src/test/java/test/io/smallrye/openapi/runtime/scanner/jakarta/RolesAllowedResource1.java
+++ b/extension-jaxrs/src/test/java/test/io/smallrye/openapi/runtime/scanner/jakarta/RolesAllowedResource1.java
@@ -8,6 +8,8 @@ import jakarta.ws.rs.Path;
 import jakarta.ws.rs.Produces;
 import jakarta.ws.rs.core.Response;
 
+import org.eclipse.microprofile.openapi.annotations.security.SecurityRequirement;
+
 @Path(value = "/v1")
 @RolesAllowed(value = "admin")
 @SuppressWarnings(value = "unused")
@@ -16,6 +18,7 @@ public class RolesAllowedResource1 {
     @GET
     @Path(value = "secured")
     @Produces(value = "application/json")
+    @SecurityRequirement(name = "rolesScheme")
     public Response getSecuredData(int id) {
         return null;
     }

--- a/extension-jaxrs/src/test/java/test/io/smallrye/openapi/runtime/scanner/javax/RolesAllowedResource1.java
+++ b/extension-jaxrs/src/test/java/test/io/smallrye/openapi/runtime/scanner/javax/RolesAllowedResource1.java
@@ -8,6 +8,8 @@ import javax.ws.rs.Path;
 import javax.ws.rs.Produces;
 import javax.ws.rs.core.Response;
 
+import org.eclipse.microprofile.openapi.annotations.security.SecurityRequirement;
+
 @Path(value = "/v1")
 @RolesAllowed(value = "admin")
 @SuppressWarnings(value = "unused")
@@ -16,6 +18,7 @@ public class RolesAllowedResource1 {
     @GET
     @Path(value = "secured")
     @Produces(value = "application/json")
+    @SecurityRequirement(name = "rolesScheme")
     public Response getSecuredData(int id) {
         return null;
     }

